gcs/src/it/scala/org/broadinstitute/monster/storage/gcs/GcsApiIntegrationSpec.scala

```diff
@@ -266,4 +265,241 @@ class GcsApiIntegrationSpec
 
     err.left.value.getMessage should include("404")
   }
+
+  // deleteObject
+  private def gcsExists(blob: BlobId): Boolean =
+    Option(gcsClient.get(blob)).exists(_.exists())
+
+  it should "delete a GCS object and return true upon success" in {
+    val wasDeleted = writeGzippedTestFile.use { blob =>
+      withClient { api =>
+        api.deleteObject(blob.getBucket, blob.getName).map { reportedDelete =>
+          !gcsExists(blob) && reportedDelete
+        }
+      }
+    }
+
+    wasDeleted.unsafeRunSync() shouldBe true
+  }
+
+  it should "return false if deleting a GCS object that doesn't exist" in {
+    val wasDeleted = withClient(_.deleteObject(bucket, "foobar"))
+    wasDeleted.unsafeRunSync() shouldBe false
+  }
+
+  // createObject
+  private val textPlain = `Content-Type`(MediaType.text.`plain`)
+
+  it should "create an object in GCS in one upload with no expected md5" in {
+    val path = s"test/${OffsetDateTime.now()}/foobar"
+
+    val wasCreated = withClient { api =>
+      api
+        .createObject(bucket, path, textPlain, None, Stream.emits(bodyText.getBytes))
+        .bracket(_ => IO.delay(gcsExists(BlobId.of(bucket, path)))) { _ =>
+          IO.delay(gcsClient.delete(bucket, path)).as(())
+        }
+    }
+
+    wasCreated.unsafeRunSync() shouldBe true
+  }
+
+  it should "create an object in GCS in one upload with a correct expected md5" in {
+    val path = s"test/${OffsetDateTime.now()}/foobar"
+
+    val wasCreated = withClient { api =>
+      api
+        .createObject(
+          bucket,
+          path,
+          textPlain,
+          Some(bodyMd5),
+          Stream.emits(bodyText.getBytes)
+        )
+        .bracket(_ => IO.delay(gcsExists(BlobId.of(bucket, path)))) { _ =>
+          IO.delay(gcsClient.delete(bucket, path)).as(())
+        }
+    }
+
+    wasCreated.unsafeRunSync() shouldBe true
+  }
+
+  it should "fail to create an object in GCS in one upload with an incorrect md5" in {
+    val path = s"test/${OffsetDateTime.now()}/foobar"
+
+    val tryCreate = withClient { api =>
+      api.createObject(
+        bucket,
+        path,
+        textPlain,
+        Some(DigestUtils.md5Hex("abcdef1234567890")),
+        Stream.emits(bodyText.getBytes)
+      )
+    }
+
+    tryCreate.recover {
+      case GcsApi.GcsFailure(status, body, _) =>
+        status shouldBe Status.BadRequest
+        body should include("MD5")
+        ()
+    }.unsafeRunSync()
+
+    gcsExists(BlobId.of(bucket, path)) shouldBe false
+  }
+
+  it should "should check if a GCS object exists and return true with an md5" in {
```

Comment:
Don't need "should" at the beginning of the test string for this test and the next one